### ISSUE TITLE
Translating embedded CMB2 with GlotPress

### ISSUE THREAD

**Issue by justnorris** [CMB2/CMB2#967]
My WordPress.org plugin is using CMB2 to handle the admin pages  - I've embedded CMB2 inside my plugin.

Now when I look at the strings at Glotpress ( https://translate.wordpress.org/ ) - it's a big mumbo-jumbo over there because CMB strings are mixed up with my plugin strings.

I searched around, and it seems that there is nothing I can do about that from the translate.wordpress.org side - the only viable solution would be to remove CMB2 from my code. But that's not an option for me at the moment as well. 

Has anyone encountered this issue? Any ideas how to solve it?

**Comment by tw2113** [comment]
Hi @justnorris 

My biggest question is which part are you trying to solve? Ensuring that CMB2 is translated to the proper language? Or trying to make sure that its strings aren't lumped together with yours?

Also a link to the actual glotpress listing would help us visualize what you're seeing.

**Comment by justnorris** [comment]
Hi @tw2113 

The plugin I'm talking about is Easy Photography portfolio.
Photography Portfolio: https://wordpress.org/plugins/photography-portfolio/
Translation page: https://translate.wordpress.org/projects/wp-plugins/photography-portfolio


Yes to both your questions.
It kind of defeats the purpose of translations, if only photography-portfolio part is translated. I think that's a no-brainer.

What I'm trying to achieve is - when people ask me that they want to translate the plugin - I could send them to Wordpress translation project, and they'd have to only translate those few strings that my plugin has, instead of having to dig through 15 pages of strings to find the photography portfolio lines.

**Comment by tw2113** [comment]
Throwing some thoughts out there, I'm betting glotpress gathers ALL of the textdomains it can find, and goes from there, regardless of any match status. Not something we're going to be able to amend.

On the other hand, you could use something like http://tgmpluginactivation.com/ to decouple CMB2 from your plugin directly, and instead make it a required 2nd plugin, being installed from its own repo on wordpress.org. As long as you're wrapping the spots that use it in proper actions/callbacks, it shouldn't whitescreen the sites if they don't have it installed yet. It would remove the strings from your plugin's glotpress list. I do understand the fact that it would be a bit of unexpected work on the plugin user's side because they suddenly need to install this 2nd thing that used to come with it already. Really depends on how you decide to handle things.

**Comment by justnorris** [comment]
> Throwing some thoughts out there, I'm betting glotpress gathers ALL of the textdomains it can find, and goes from there, regardless of any match status. 
Yeah, unfortunately they do that. I wish there was a way to tell glotpress only to use a certain textdomain for a plugin.

> you could use something like http://tgmpluginactivation.com/ to decouple CMB2 
I had the same thought, but that's just not going to look right for the user. I guess I'll have to endure the pile of translation strings in the glotpress repo, in hopes that someday they decide to update it a little more. Or maybe one day my plugin matures enough for me to build my own dashboard that doesn't rely on CMB2.
